src/hooks/useSelectInterest.js

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+
+export default function useSelectInterest(api) {
+  const [availableTopics, setAvailableTopics] = useState([]);
+  const [subscribedTopics, setSubscribedTopics] = useState([]);
+  const [allTopics, setAllTopics] = useState([]);
+  const [subscribedSearches, setSubscribedSearches] = useState([]);
+  const [showingSpecialInterestModal, setshowingSpecialInterestModal] =
+    useState(false);
+
+  useEffect(() => {
+    api.getAvailableTopics((data) => {
+      setAvailableTopics(data);
+    });
+
+    api.getSubscribedTopics((data) => {
+      setSubscribedTopics(data);
+    });
+
+    //custom interest filters
+    api.getSubscribedSearchers((data) => {
+      setSubscribedSearches(data);
+    });
+  }, [api]);
+
+  useEffect(() => {
+    let newAllTopics = [...availableTopics, ...subscribedTopics];
+    newAllTopics.sort((a, b) => a.title.localeCompare(b.title));
+    setAllTopics(newAllTopics);
+  }, [availableTopics, subscribedTopics]);
+
+  function subscribeToTopic(topic) {
+    setSubscribedTopics([...subscribedTopics, topic]);
+    setAvailableTopics(availableTopics.filter((each) => each.id !== topic.id));
+    api.subscribeToTopic(topic);
+  }
+
+  function unsubscribeFromTopic(topic) {
+    setSubscribedTopics(
+      subscribedTopics.filter((each) => each.id !== topic.id),
+    );
+    setAvailableTopics([...availableTopics, topic]);
+    api.unsubscribeFromTopic(topic);
+  }
+
+  function toggleTopicSubscription(topic) {
+    if (subscribedTopics.includes(topic)) {
+      unsubscribeFromTopic(topic);
+    } else {
+      subscribeToTopic(topic);
+    }
+  }
+
+  //subscribe to custom interest filter
+  function subscribeToSearch(response) {
+    api.subscribeToSearch(response, (data) => {
+      setSubscribedSearches([...subscribedSearches, data]);
+    });
+  }
+
+  //remove custom interest filter
+  function removeSearch(search) {
+    console.log("unsubscribing from search" + search);
+    setSubscribedSearches(
+      subscribedSearches.filter((each) => each.id !== search.id),
+    );
+    api.unsubscribeFromSearch(search);
+  }
+
+  function isSubscribed(topic) {
+    return subscribedTopics
+      .map((subscribedTopic) => subscribedTopic.id)
+      .includes(topic.id)
+      ? ""
```

Comment:
`isSubscribed` should return a boolean, by the nature of its name. If you want to return a string, it's cool, and works well for us, but then: 
1. function should probably be named `subscriptionStatus` or similar
2. it should return either `"subscribed"` or `"unsubscribed"`. Empty string does not make sense as a return for the subscribed state. See what I mean?

Reply:
That's a good point! I have decided to keep the `isSubscribed` name and turn it into a **boolean** so that it is independent of possible future CSS class selector name changes of tags.